src/openms/source/CHEMISTRY/IsotopeDistribution.cpp

```diff
@@ -254,6 +254,13 @@ namespace OpenMS
       }
   }
 
+  void IsotopeDistribution::calcFragmentIsotopeDist(const IsotopeDistribution& fragment_isotope_dist, const IsotopeDistribution& comp_fragment_isotope_dist, const std::vector<UInt>& precursor_isotopes)
+  {
+    ContainerType result;
+    calcFragmentIsotopeDist_(result, fragment_isotope_dist.distribution_, comp_fragment_isotope_dist.distribution_, precursor_isotopes);
+    distribution_ = result;
```

Comment:
Honestly, I don't know haha. I followed the pattern that was used for the other extremely similar functions in that file. It does look like an example of strong exception safety though.